apps/shell/tash_security.c

```diff
@@ -49,26 +69,15 @@ void tash_check_security(int fd)
 		line_buff = tash_read_input_line(fd);
 		len = strlen(line_buff);
 
-		sprintf((char *)sha256, "%s", CONFIG_TASH_PASSWORD_SHA256);
-
-		for (i = 0; i < 64; i++) {
-			strncpy(chr, (char *)&sha256[i], 1);
-			temp[i % 2] = strtoul(chr, NULL, 16);
-
-			if (i % 2) {
-				sha256_hex[i / 2] = (temp[0] << 4) | temp[1];
-			}
-		}
-
-		mbedtls_sha256((unsigned char *)line_buff, len, result, 0);
+		mbedtls_sha256((unsigned char *)line_buff, len, input_sha256_hex, 0);
 
 		tash_free(line_buff);
 
-		if (!strncmp((char *)sha256_hex, (char *)result, 32)) {
+		if (!strncmp((char *)pw_sha256_hex, (char *)input_sha256_hex, 32)) {
 			break;
 		} else if (len != 0) {
```

Comment:
It is better to move this `len!=0` to line 71 and change `else` here. How about this?